### ISSUE TITLE
No result provided from search() method

### ISSUE THREAD

**Issue by Booteille** [neet/masto.js#632]
Hi.

I am trying to get results from masto.search() but I get no result under accounts/hashtags/statuses returned by the function.

Here is the code:

```ts
const r = await masto.search({
    q: "booteille"
  }).next()

  if (r) {
    console.log(r.done)
    console.log(r.value)
  }
```

Here are console logs:
```console
false
Object { accounts: [], statuses: [], hashtags: [] }
```

Did I do something wrong? I tried with 3 different keywords, I have results for each with the user logged in on Mastodon interface.
App has global read permissions.

Thanks.

**Comment by neet** [comment]
Thank you for reporting the issue. Unfortunately, the code you provided has worked without problem in my instance.

Could you try the exact same keyword in the Web interface and verify with a network inspector that the request to `/api/v2/search` returns the results you expect? The `masto.search` method corresponds to the search bar in the very left column.

<img width="1117" alt="image" src="https://user-images.githubusercontent.com/19276905/174821447-85768c0b-dca3-46b8-8fb1-b357161fb82f.png">

I'm not very familiar with how WebFinger works, but since Web UI always specifies `resolve` param described in [this page](https://docs.joinmastodon.org/methods/search/), adding `resolve: true` would also help you get the same result as the Web UI.


```diff
const r = await masto.search({
    q: "booteille"
+    resolve: true
  }).next()
```

**Comment by Booteille** [comment]
Hi!

Thanks for your fast answer.

Unfortunately, using resolve did not work neither. The API does not seem to return wanted results.


![Capture d’écran du 2022-06-22 14-32-23](https://user-images.githubusercontent.com/2203721/175029978-1f601901-dc14-4994-893b-128f95f0c2ef.png)

After investigating, I found the request does not seem to contain the `q` parameter (neither the resolve one, when I add it).

Editing through Firefox dev tools the request to add the parameter returns results as intented.

Any idea why it's not working on my case?

**Comment by neet** [comment]
I believe this issue is fixed by #700. If you think you still have a problem, feel free to reopen it.
